packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/BaseViewManager.java

```diff
@@ -40,7 +40,7 @@
  * provides support for base view properties such as backgroundColor, opacity, etc.
  */
 public abstract class BaseViewManager<T extends View, C extends LayoutShadowNode>
-    extends ViewManager<T, C> implements BaseViewManagerInterface<T> {
+    extends ViewManager<T, C> implements BaseViewManagerInterface<T>, View.OnLayoutChangeListener {
```

Comment:
layout on a view is updated [here](https://github.com/facebook/react-native/blob/2b93491779073d7a97e4826300d5c301c835d6d9/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/fabric/mounting/SurfaceMountingManager.java#L1002). We can use [onSizeChanged callback](https://github.com/facebook/react-native/blob/2b93491779073d7a97e4826300d5c301c835d6d9/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewGroup.java#L480) from ReactViewGroup instead of a listener here but that makes BaseViewManager dependent upon ReactViewGroup. i.e. we'll have to typecast generic `T` view to ReactViewGroup to achieve the same.